Mathlib/Data/Sign.lean

```diff
@@ -250,6 +250,18 @@ instance : CoeTC SignType α :=
 
 -- Porting note: `cast_eq_coe` removed, syntactic equality
 
+/-- Casting out of `SignType` respects composition with functions preserving `0, 1, -1`. -/
+lemma comp_cast' {β : Type*} [One β] [Neg β] [Zero β]
+    (f : α → β) (h₁ : f 1 = 1) (h₂ : f 0 = 0) (h₃ : f (-1) = -1) (s : SignType) :
+    f s = s := by
+  cases s <;> simp only [SignType.cast, h₁, h₂, h₃]
+
+/-- Casting out of `SignType` respects composition with suitable bundled homomorphism types. -/
+lemma comp_cast {α β F : Type*} [AddGroupWithOne α] [One β] [SubtractionMonoid β]
```

Comment:
This should be called `map_cast`, and I guess the above can be renamed similarly.

Reply:
OK, will do. There are also a bunch of pre-existing lemmas about `SignType.cast` with names like `coe_one`, should these become `cast_one` etc? (I am not sure if there is supposed to be a distinction between casting and coercion, or whether one is just the mathlib4 equivalent of the other.)